tools/clang_tidy/lib/clang_tidy.dart

```diff
@@ -171,23 +189,91 @@ class ClangTidy {
     return repo.changedFiles;
   }
 
-  /// Given a build commands json file, and the files with local changes,
-  /// compute the lint commands to run.
-  @visibleForTesting
-  Future<List<Command>> getLintCommandsForChangedFiles(
-    List<dynamic> buildCommandsData,
-    List<io.File> changedFiles,
-  ) async {
-    final List<Command> buildCommands = <Command>[];
-    for (final dynamic data in buildCommandsData) {
-      final Command command = Command.fromMap(data as Map<String, dynamic>);
-      final LintAction lintAction = await command.lintAction;
-      // Short-circuit the expensive containsAny call for the many third_party files.
-      if (lintAction != LintAction.skipThirdParty && command.containsAny(changedFiles)) {
-        buildCommands.add(command);
+  /// Returns f(n) = value(n * [shardCount] + [id]).
+  Iterable<T> _takeShard<T>(Iterable<T> values, int id, int shardCount) sync* {
+    int count = 0;
+    for (final T val in values) {
+      if (count % shardCount == id) {
+        yield val;
+      }
+      count++;
+    }
+  }
+
+  /// This returns a `_SetStatusCommand` for each [Command] in [items].
+  /// `Intersection` if the Command shows up in [items] and its filePath in all
+  /// [filePathSets], otherwise `Difference`.
+  Iterable<_SetStatusCommand> _calcIntersection(
```

Comment:
The following is more idiomatic. You can use `Iterable.every` in `allSetsContain`, and you can use a "collection-for" instead of a `sync*`:
```dart
Iterable<_SetStatusCommand> _calcIntersection(
  Iterable<Commands> items,
  Iterable<Set> filePathSets,
) {
  bool allSetsContain(Command command) {
    return filePathSets.every(
      (Set<String> filePathSet) => filePathSet.contains(command.filePath),
    );
  }
  return <_SetStatusCommand>[
    for (final Command command in items)
      if (allSetsContain(command))
        _SetStatusCommand(_SetStatus.Intersection, command)
      else
        _SetStatusCommand(_SetStatus.Difference, command),
  ];
}
```